cluster/vagrant/provision-master.sh

```diff
@@ -21,16 +21,16 @@ set -e
 # See: https://github.com/mitchellh/vagrant/issues/2430
 hostnamectl set-hostname ${MASTER_NAME}
 
-# Workaround to vagrant inability to guess interface naming sequence
-# Tell system to abandon the new naming scheme and use eth* instead
-rm -f /etc/sysconfig/network-scripts/ifcfg-enp0s3
-
 # Disable network interface being managed by Network Manager (needed for Fedora 21+)
 NETWORK_CONF_PATH=/etc/sysconfig/network-scripts/
-grep -q ^NM_CONTROLLED= ${NETWORK_CONF_PATH}ifcfg-eth1 || echo 'NM_CONTROLLED=no' >> ${NETWORK_CONF_PATH}ifcfg-eth1
-sed -i 's/^#NM_CONTROLLED=.*/NM_CONTROLLED=no/' ${NETWORK_CONF_PATH}ifcfg-eth1
+for if_conf in ${NETWORK_CONF_PATH}ifcfg-*; do
+  grep -q ^NM_CONTROLLED= ${if_conf} || echo 'NM_CONTROLLED=no' >> ${if_conf}
+  sed -i 's/#^NM_CONTROLLED=.*/NM_CONTROLLED=no/' ${if_conf}
+done;
 systemctl restart network
```

Comment:
On VirtualBox, this command fails.

I don't get why you are disabling NetworkManager for all the interfaces.

On VirtualBox, you are editing the following:

```
/etc/sysconfig/network-scripts/ifcfg-enp0s3
/etc/sysconfig/network-scripts/ifcfg-eth1
/etc/sysconfig/network-scripts/ifcfg-lo
```

For each of the following I am getting files like the following:

```
[root@kubernetes-master vagrant]# cat /etc/sysconfig/network-scripts/ifcfg-enp0s3 
# Generated by dracut initrd
DEVICE="enp0s3"
ONBOOT=yes
NETBOOT=yes
UUID="92af5668-dde0-46ea-9c1f-4508af6cdae2"
IPV6INIT=yes
BOOTPROTO=dhcp
TYPE=Ethernet
NAME="enp0s3"
NM_CONTROLLED=no
NM_CONTROLLED=no
[root@kubernetes-master vagrant]# cat /etc/sysconfig/network-scripts/ifcfg-eth1
#VAGRANT-BEGIN
# The contents below are automatically generated by Vagrant. Do not modify.
NM_CONTROLLED=no
BOOTPROTO=none
ONBOOT=yes
IPADDR=10.245.1.2
NETMASK=255.255.255.0
DEVICE=eth1
PEERDNS=no
#VAGRANT-END
NM_CONTROLLED=no
[root@kubernetes-master vagrant]# cat /etc/sysconfig/network-scripts/ifcfg-lo
DEVICE=lo
IPADDR=127.0.0.1
NETMASK=255.0.0.0
NETWORK=127.0.0.0
# If you're having problems with gated making 127.0.0.0/8 a martian,
# you can change this to something else (255.255.255.255, for example)
BROADCAST=127.255.255.255
ONBOOT=yes
NAME=loopback
NM_CONTROLLED=no
NM_CONTROLLED=no
```